### DISCUSSION TITLE
The controller is not available at localhost:9990

### DISCUSSION THREAD

**Discussion by Flyslinger2** [Keyfactor/ejbca-ce#310] [Q&A]
[root@localhost opt]# /opt/wildfly/bin/jboss-cli.sh --connect '/subsystem=elytron/credential-store=defaultCS:add(location=keystore/credentials, relative-to=jboss.server.config.dir, credential-reference={clear-text="{EXT}/usr/bin/wildfly_pass", type="COMMAND"}, create=true)'

Failed to connect to the controller: The controller is not available at localhost:9990: java.net.ConnectException: WFLYPRT0053: Could not connect to remote+http://localhost:9990. The connection failed: WFLYPRT0053: Could not connect to remote+http://localhost:9990. The connection failed: Connection refused

**Comment by primetomas** [comment]
"java.net.ConnectException" means it can't eatablish a network connection. You may have missed to configure remoting on your WildFly server. Check the documentation. Also check the WildFly server log to see that it has started correctly. It may have been some error with an earlier command that caused it to go down.
